lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,29 +1,921 @@
+                raise Exception('CBVs could not be loaded. CBVCorrector must exit')
+
+        # Align the CBVs with the lightcurve flux using the cadence numbers
+        for idx in np.arange(len(cbvs)):
+            self.lc = cbvs[idx].align(self.lc, trim_lc=True)
+
+        self.cbvs = cbvs
+
+
+    def correct(self, alpha=1e-20, l1_ratio=0.01, ext_dm=None, cadence_mask=None):
+        """ Performs the correction using RegressionCorrector methods
+
+        This method will assemble the full design matrix collection composed of
+        cbv_design_matrix and extra_design_matrix. Then use the super-class
+        RegressionCorrector.correct to perform the correction
+
+        TODO: A whole bunch! This is a shell of a method. Just to demonstrate
+        basic functionality. We should consider allowing for standard
+        regaulrization methods such as Ridge Regression (L2 Norm) and Lasso (L1
+        Norm).  Maybe even BIC or AIC methods also.
+
+        Eventually, the plan is to use over-fitting and under-fitting goodness
+        metrics to constrain the regression fit.
+
+        Parameters
+        ----------
+        ext_dm  :  `.DesignMatrix` or `.DesignMatrixCollection`
+            Optionally pass an extra design matrix to also be used in the fit
+        cadence_mask : np.ndarray of bools (optional)
+            Mask, where True indicates a cadence that should be used.
+
+        """
+
+        if cadence_mask is None:
+            self.cadence_mask = np.ones(len(self.lc.time), bool)
+        else:
+            self.cadence_mask = np.copy(cadence_mask)
+
+        # Create the design matrix collection with CBVs, plus extra passed basis vectors
+
+        # If any DesignMatrix was passed then store it
+        self.extra_design_matrix = ext_dm
+
+        #Create a CBV design matrix for each CBV set loaded
+        self.cbv_design_matrix = []
+        for idx in np.arange(len(self.cbvs)):
+            cbv_index_names = [cbv_index for cbv_index in self.cbv_indices[idx]]
+            self.cbv_design_matrix.append(DesignMatrix(self.cbvs[idx]._cbvs_to_matrix(),
+                    name=self.cbvs[idx].cbv_type, columns=cbv_index_names))
+
+        # Create the full design matrix collection
+        if self.extra_design_matrix is not None:
+            dm_to_flatten = [[cbv_dm for cbv_dm in self.cbv_design_matrix], [self.extra_design_matrix]]
+            flattened_dm_list = [item for sublist in dm_to_flatten for item in sublist]
+        else:
+            dm_to_flatten = [[cbv_dm for cbv_dm in self.cbv_design_matrix]]
+            flattened_dm_list = [item for sublist in dm_to_flatten for item in sublist]
+        self.design_matrix_collection = DesignMatrixCollection(flattened_dm_list)
+        self.design_matrix_collection._validate()
+
+            
+        # Use RegressionCorrector.correct for the actual fitting
+       #super(CBVCorrector, self).correct(design_matrix)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [indentation is not a multiple of four (comment)](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4896575)